### ISSUE TITLE
catch and GoogleTest throw build error 

### ISSUE THREAD

**Issue by Andrea-Havron-NOAA** [Andrea-Havron-NOAA/FIMS-software#14]
**Describe the bug**
catch and GoogleTest throw build error: 

In file included from /home/runner/work/FIMS-software/FIMS-software/src/tests/catch/../../Rcpp/model.hpp:4,
                 from /home/runner/work/FIMS-software/FIMS-software/src/tests/catch/catch-model.cpp:2:
/home/runner/work/FIMS-software/FIMS-software/src/tests/catch/../../Rcpp/Common.hpp:8:[10](https://github.com/Andrea-Havron-NOAA/FIMS-software/runs/5118373008?check_suite_focus=true#step:5:10): fatal error: TMB.hpp: No such file or directory
    8 | #include <TMB.hpp>

**Expected behavior**
GHA needs to know the location of TMB,hpp so tests do not fail on build

**Additional context**
@Bai-Li-NOAA and @msupernaw, do we need to set up a Makefile with a build command such as [here](https://github.com/pbs-assess/sdmTMB/blob/main/Makefile) or [here](https://github.com/glmmTMB/glmmTMB/blob/master/Makefile)?

**Comment by Bai-Li-NOAA** [comment]
I think the issue will be resolved if you comment out [#define TMB_model](https://github.com/Andrea-Havron-NOAA/FIMS-software/blob/a52e4240b68f46fa17d478868b3f0f7327ac14f0/src/Rcpp/Common.hpp#L6). Please see the successful GitHub actions run logs [here](https://github.com/Bai-Li-NOAA/FIMS-software/runs/4488984813?check_suite_focus=true) and the [common.hpp](https://github.com/Bai-Li-NOAA/FIMS-software/blob/tests/src/Rcpp/Common.hpp). In the [CMakeLists.txt](https://github.com/Andrea-Havron-NOAA/FIMS-software/blob/master/src/tests/googletest/CMakeLists.txt), STD_LIB under target_compile_definitions helps compile the model without including `#define TMB_model`.

I also noticed that there are some updates in model.hpp (e.g. removing theta), you may need to update the the cpp files under src/tests as well to be able to test the new model.

Hope this helps!

**Comment by Andrea-Havron-NOAA** [comment]
Thanks for the suggestions @Bai-Li-NOAA! I first moved #define TMB_MODEL to the top of the .cpp file and got the same error on GHA. I've tried commenting out and compiling with `flags="-DTMB_MODEL -w" ` and this works on my local machine. I will try pushing to gitHub to see if this fixes the GHA errors.

I have spotted a related issue that will be a problem when we move to TMB's new AD framework. The model compiles successfully when #define TMB_MODEL is specified at the top of the .cpp file but fails when it is commented out and compiled with `framework = "TMBad"`  and `flags="-DTMB_MODEL -w" ` with a `file too big error` 

To summarize the following fails to compile (using the newest development version of TMB from gitHub):
`TMB::compile('src/logisticGrowth.cpp', flags="-DTMB_MODEL -w", framework = 'TMBad') `(with #define TMB_MODEL commented out in logisticGrowth.cpp)

The following successfully compiles (using the newest development version of TMB from gitHub):
`TMB::compile('src/logisticGrowth.cpp', framework = 'TMBad') ` (with #define TMB_MODEL  not commented out in logisticGrowth.cpp)
`TMB::compile('src/logisticGrowth.cpp', flags="-DTMB_MODEL -w", framework = 'CppAD') ` (with #define TMB_MODEL commented out in logisticGrowth.cpp)

**Comment by Bai-Li-NOAA** [comment]
Comment out #define TMB_model in the common.hpp instead of a .cpp file may help. I suspect there will be more errors because of other updates in model.hpp. Would you like to set up a meeting and set up the testing framework locally and debug together?

**Comment by Bai-Li-NOAA** [comment]
The Google Test workflow works now! I suggest that you remove the [catch workflow](https://github.com/Andrea-Havron-NOAA/FIMS-software/blob/master/.github/workflows/catch.yml) and only use the Google Test workflow.

**Comment by Andrea-Havron-NOAA** [comment]
removed catch workflow
